specs/IsNonClientRegionSupportEnabled.md

```diff
@@ -0,0 +1,93 @@
+# Background
+
+To improve the developer experience for customizing non-client regions, WebView2 is working to support using DOM elements as non-client regions. We currently have limited support for title bar aka 
+draggable regions and are working on building out support for caption controls and resize regions. 
+
+For security and flexibility, we want developers to be able to enable or disable all custom non-client functionality per WebView. Non-client functionality will affect that app window’s size and position so 
+it’s important that developers can definitively toggle access. This can be achieved in a limited way using a feature flag, but feature flags are applied per WebView2 environment, thus, an API on the 
+WebView2 to enable/disable non-client support via a setting is the better solution.
+
+# Description
+`IsNonClientRegionSupportEnabled` defaults to `FALSE`. Disabling/Enabling `IsNonClientRegionSupportEnabled` takes effect after the next navigation. Currently, draggable regions is the only non-client 
+experience we have implemented. Eventually, this setting will expand to enable other non-client functionality, such as resize and caption controls. 
+
+When the setting is set to `TRUE`, then the following non-client region support for the top level document will be enabled:  
+
+* Web pages will be able to use the `app-region` CSS style. 
+
+* Draggable Regions declared with the CSS style `app-region: drag` will support title bar functionality, for example, the dragging of the app window, the title bar context menu upon right click, and the maximizing/ restoring of the window size upon double click of the html element. 
+
+When set to `FALSE`, then all non-client region support will be disabled.  
+
+* Web pages will not be able to use the `app-region` CSS style. 
+
+# Examples
+```cpp
+wil::com_ptr<ICoreWebView2> webView;
+void SettingsComponent::ToggleNonClientRegionSupportEnabled()
+{
+    // Get webView's current settings
+    wil::com_ptr<ICoreWebView2Settings> coreWebView2Settings;
+    CHECK_FAILURE(webView->get_Settings(&coreWebView2Settings));
+    wil::com_ptr<ICoreWebView2Settings9> coreWebView2Settings9;
+    coreWebView2Settings9 = coreWebView2Settings.try_query<ICoreWebView2Settings9>();
+    if (coreWebView2Settings9)
+    {
+        BOOL enabled;
+        CHECK_FAILURE(coreWebView2Settings9->get_IsNonClientRegionSupportEnabled(&enabled));
+        CHECK_FAILURE(coreWebView2Settings9->put_IsNonClientRegionSupportEnabled(!enabled));
+    }
+}
+```
+```c#
+private WebView2 _webView;
+void ToggleNonClientRegionSupportEnabled()
+{
+    var coreWebView2Settings = _webView.CoreWebView2.Settings;
+    coreWebView2Settings.IsNonClientRegionSupportEnabled = !coreWebView2Settings.IsNonClientRegionSupportEnabled;
+}
+```
+
+# Remarks
+If the feature flag for enabling the app-region CSS style is used to enable draggable regions in additional browser arguments, draggable region support will remain enabled even if the 
```

Comment:
Did we change our decision here? I thought we previously discussed honoring the feature flag as the default value for IsNonClientRegionSupportEnabled, but that if an app specifically changed the setting, then that would take priority.

Reply:
I discussed this further with Nic, and he suggested that if devs wanted the ability to toggle non-client region support then they should be using the setting not the flag. Also, using the flag would suggest that you want it to be enabled for the webview environment so it would make sense for it to override the setting.